sample-package-core/src/main/resources/OSGI-INF/custombulkaction-bulk-action-contrib.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<component name="org.nuxeo.sample.core.actions.custombulkaction">
+
+  <require>org.nuxeo.ecm.core.bulk</require>
+
+  <extension target="org.nuxeo.ecm.core.bulk" point="actions">
+    <action name="custombulkaction" bucketSize="100" batchSize="25" sequentialCommands="true"/>
+  </extension>
+
+  <extension target="org.nuxeo.runtime.stream.service" point="streamProcessor">
+    <streamProcessor name="custombulkaction" class="org.nuxeo.sample.core.CustomBulkAction"
+                     defaultConcurrency="1" defaultPartitions="1" logConfig="bulk">
```

Comment:
AFAIK, `defaultPartitions` can not be changed without deleting the topic on kafka side, this setting should be configured properly at creation if you want to have concurrent processing as it limits the number of concurrent threads.

`defaultConcurrency` limits the number of threads to start, so the following rule must be true: `defaultConcurrency <= defaultPartitions`.

It is always interesting to leave some room if you need to increase the concurrency, for instance having `defaultConcurrency="2"` `defaultPartitions ="4"`. A factor 3 could be interesting.

`1` is needed when you want to have an order.

Reply:
Thx Kevin, I've added your comments and added some example configuration from the platform to illustrate different use cases (auditLogWriter requiring order, and setProperties default and perf config)